Prox/Prox/PlaceDetails/FilterViewController.swift

```diff
@@ -0,0 +1,193 @@
+        let filter = filters[sender.tag]
+        filter.enabled = !filter.enabled
+        sender.isSelected = filter.enabled
+        delegate?.filterViewController(self, didUpdateFilters: filters)
+    }
+
+    @objc private func didPressClose() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
+        super.dismiss(animated: flag, completion: completion)
+        delegate?.filterViewController(self, didDismissWithFilters: filters)
+    }
+}
+
+extension FilterViewController: UIViewControllerAnimatedTransitioning, UIViewControllerTransitioningDelegate {
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        transitionContext.containerView.addSubview(view)
+        view.layoutIfNeeded()
+
+        UIView.animate(withDuration: 0.2, animations: {
+            if transitionContext.viewController(forKey: .to) == self {
+                self.background.alpha = 1
+                NSLayoutConstraint.deactivate(self.hideFilterConstraints)
+                NSLayoutConstraint.activate(self.showFilterConstraints, translatesAutoresizingMaskIntoConstraints: false)
+            } else {
+                self.background.alpha = 0
+                NSLayoutConstraint.deactivate(self.showFilterConstraints)
+                NSLayoutConstraint.activate(self.hideFilterConstraints, translatesAutoresizingMaskIntoConstraints: false)
+            }
+
+            self.view.layoutIfNeeded()
+        }, completion: { _ in
+            transitionContext.completeTransition(true)
+        })
+    }
+
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return 0.2
+    }
+
+    func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        return self
+    }
+
+    func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        return self
+    }
+}
+
+fileprivate class FilterButton: InsetButton {
+    override init() {
+        super.init()
+
+        titleLabel?.font = UIFont.systemFont(ofSize: 16)
+        contentEdgeInsets = UIEdgeInsets(top: 8, left: 0, bottom: 8, right: 16)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: 20, bottom: 0, right: 0)
+        layer.borderWidth = 1
+        layer.borderColor = Colors.filterButtonBorder.cgColor
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    fileprivate override var isSelected: Bool {
+        didSet {
+            super.isSelected = isSelected
+            if isSelected {
+                setTitleColor(Colors.filterButtonCheckedForeground, for: .normal)
+                setImage(#imageLiteral(resourceName: "icon_selected"), for: .normal)
```

Comment:
nit: does this reload the image from disk every time? If we see perf issues, we might want to consider caching it.